xrootd/client/filesystem.go

```diff
@@ -25,3 +26,14 @@ func (fs *FileSystem) Dirlist(ctx context.Context, path string) ([]protocol.Entr
 	}
 	return resp.Entries, err
 }
+
+// Open returns the file handle for a file together with the compression and the stat info.
+func (fs *FileSystem) Open(ctx context.Context, path string, mode open.Mode, options open.Options) (protocol.FileHandle, *protocol.FileCompression, *protocol.EntryStat, error) {
```

Comment:
looking at the returned types, I have the urge to merge them into a single type, `File`, which would eventually implement `io.Reader`, `io.Writer`, `io.Closer`, ... with 2 additional methods:
- one to return this `FileCompression`
- the other to return `os.FileInfo` that our `EntryStat` implements already.

but perhaps it's a bit too soon to introduce such "high-level" abstractions? (or the wrong "level"?)

still... this really "feels" right:
```go
func (fs *FileSystem) Open(ctx context.Context, 
     path string, mode open.Mode, options open.Options) 
     (xrootd.File, error) { ... }
```

Reply:
I gave it a try. From my point of view, it looks fine, but I will be happy to hear your opinion. :)